patroni/dcs/etcd.py

```diff
@@ -264,8 +248,11 @@ def _get_machines_cache_from_srv(self, srv):
 
     def _get_machines_cache_from_dns(self, host, port):
         """One host might be resolved into multiple ip addresses. We will make list out of it"""
-        ret = ['{0}://{1}:{2}'.format(self.protocol, ip, port) for ip in set(self._dns_resolver.resolve(host))]
-        return list(set(ret)) if ret else ['{0}://{1}:{2}'.format(self.protocol, host, port)]
+        if self.protocol == 'http':
+            ret = ['{0}://{1}:{2}'.format(self.protocol, *r[4]) for r in self._dns_resolver.resolve(host, port)]
```

Comment:
According to https://docs.python.org/3/library/socket.html#socket.getaddrinfo and https://docs.python.org/2/library/socket.html#socket.getaddrinfo the Python tuple returned for the AF_INET6 contains 4 fields (host, port, flow_info and scope_id).